source/app/models/url.rb

```diff
@@ -0,0 +1,12 @@
+class Url < ActiveRecord::Base
+  before_save :sanitize_address
+  before_save :compress_id
+
+  def sanitize_address 
+    self.address.gsub!(/https:\/\/|http:\/\//,'')
+  end
+
+  def compress_id
+    self.id = IdCompressor.compress(Url.count + 1)
```

Comment:
`.count` will result in a db call for each attempt to save a url record. you may want to look into counter_cache or an alternative method for generating short urls